src/Time/BoundaryHistory.hpp

```diff
@@ -123,7 +123,8 @@ class BoundaryHistory {
   // We use pointers instead of iterators because deque invalidates
   // iterators when elements are inserted or removed at the ends, but
   // not pointers.
-  std::map<std::pair<const Time*, const Time*>, CouplingResult> coupling_cache_;
+  mutable std::map<std::pair<const Time*, const Time*>, CouplingResult>
```

Comment:
are there any thread safety issues related to this map?

Reply:
Right, this should never be accessed by multiple threads at once.